solutions/devsprint-leonardo-santos-1/FinanceAppTests/Service/FinanceServiceTests.swift

```diff
@@ -0,0 +1,62 @@
+import XCTest
+@testable import FinanceApp
+
+class FinanceServiceTests: XCTestCase {
+
+    func testFetchUserProfile() {
+        //given
+        let networkSpy = NetworkSpy()
+        let financeService = FinanceService(networkClient: networkSpy)
+
+        //when
+        financeService.fetchUserProfile { _ in
+
+            //then
+            XCTAssertEqual(networkSpy.performRequestCount, 1)
+        }
+    }
+
+    func testFetchUserProfileReturning() {
+        //given
+        let networkMock = NetworkMock()
+        let financeService = FinanceService(networkClient: networkMock)
+        let userModel = NetworkMock.getUser()
+
+        //when
+        financeService.fetchUserProfile { data in
+
+            //then
+            XCTAssertEqual(userModel, data)
+        }
+    }
+
+}
+
+class NetworkSpy: NetworkClientProtocol {
+    var performRequestCount = 0
```

Comment:
Deixo como dica utilizar sempre `private(set)` nas suas propriedades que NÃO podem ser manipuladas externamente durante os testes. O motivo disso é que se algum desenvolvedor quiser colocar `performRequestCount = 1` nos testes, isso forçará o teste a dar certo. Lembre-se os testes tem que validarem a si mesmos

```suggestion
    private(set) var performRequestCount = 0
```